skimage/transform/_warps_cy.pyx

```diff
@@ -119,37 +143,126 @@ def _warp_fast(np_floats[:, :] image, np_floats[:, :] H, output_shape=None,
     would be [0, 1, 2, 1, 0, 1, 2].
 
     """
+    cdef np.intp_t[2] shape
+    if output_shape is None:
+        shape[0] = np.PyArray_DIM(image, 0)
+        shape[1] = np.PyArray_DIM(image, 1)
+    else:
+        shape[0] = output_shape[0]
+        shape[1] = output_shape[1]
 
-    cdef np_floats[:, ::1] img = np.ascontiguousarray(image)
-    cdef np_floats[:, ::1] M = np.ascontiguousarray(H)
+    cdef np_floats[:, ::1] img = np.PyArray_GETCONTIGUOUS(image)
```

Comment:
Why are you using the capi?

Why not simply leave this as is, I don't think you are avoiding copies here

https://github.com/numpy/numpy/blob/v1.17.0/numpy/core/_asarray.py#L179